rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/moderationrequest/Sw360ModerationRequestService.java

```diff
@@ -193,10 +240,87 @@ public Map<PaginationData, List<ModerationRequest>> getRequestsByState(User sw36
      */
     public ModerationState acceptRequest(ModerationRequest request, String moderatorComment, @NotNull User reviewer)
             throws TException {
+        ComponentService.Iface sw360ComponentClient = getThriftComponentClient();
+        ProjectService.Iface sw360ProjectClient = getThriftProjectClient();
+        LicenseService.Iface sw360LicenseClient = getThriftLicenseClient();
+        SPDXDocumentService.Iface sw360SPDXClient = getThriftSPDXDocumentClient();
+        DocumentCreationInformationService.Iface documentCreationInfoClient = getThriftDocumentCreationInfo();
+        PackageInformationService.Iface packageInfoClient = getThriftPackageInfo();
+        User userFromRequest = getUserFromRequest(request);
+        switch (request.getDocumentType()) {
+        case COMPONENT: {
+            if (request.isRequestDocumentDelete()) {
+                sw360ComponentClient.deleteComponent(request.getDocumentId(), reviewer);
+            } else {
+                sw360ComponentClient.updateComponentFromModerationRequest(request.getComponentAdditions(),
+                        request.getComponentDeletions(), reviewer);
+            }
+        }
+            break;
+
+        case PROJECT: {
+            if (request.isRequestDocumentDelete()) {
+                sw360ProjectClient.deleteProject(request.getDocumentId(), reviewer);
+            } else {
+                sw360ProjectClient.updateProjectFromModerationRequest(request.getProjectAdditions(),
+                        request.getProjectDeletions(), reviewer);
+            }
+        }
+            break;
+        case RELEASE: {
+            if (request.isRequestDocumentDelete()) {
+                RequestStatus deleteStatus = sw360ComponentClient.deleteRelease(request.getDocumentId(), reviewer);
+                if (deleteStatus.equals(RequestStatus.SUCCESS)) {
```

Comment:
Should the endpoint respond with failure if the `deleteStatus` is not successful?

Reply:
Addressed.